.github/workflows/maven-pr-builder.yaml

```diff
@@ -0,0 +1,29 @@
+name: Compile Java source files
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '**/*.java'
+      - 'pom.xml'
+      - 'docs/antora.yml'
+      - '.github/workflows/maven-pr-builder.yaml'
+  schedule:
+    - cron: '15 2 * * 0'
+     
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          java-version: '17'
```

Comment:
[I think this should by dynamic](https://github.com/hazelcast/hz-docs/pull/1263#discussion_r1750302258).